metaflow/datastore/task_datastore.py

```diff
@@ -553,6 +555,25 @@ def done(self):
         )
 
         if self._metadata:
+            origin_task_metadata = []
```

Comment:
You can add more assertions in https://github.com/Netflix/metaflow/blob/master/test/core/tests/resume_*.py to test this PR.

Reply:
Added Test for this too.